TicTacToe/TicTacToe.cs

```diff
@@ -27,61 +28,175 @@ public static void Main()
-            int row = int.Parse(Console.ReadLine());
-            Console.WriteLine("Enter Column:");
-            int column = int.Parse(Console.ReadLine());
+            int row = 0;
+            int column = 0;
+            //Try function in case of invalid input
+            try
+            {
+                row = Convert.ToInt32((Console.ReadLine()));
+                //Test for only 0, 1, or 2 inputs for Row
+                if (test1(row))
+                {
+                    Console.WriteLine("Enter Column:");
+
+                    column = Convert.ToInt32((Console.ReadLine()));
+                    
+                    //Test for only 0, 1, or 2 inputs for Column
+                    if (test2(column))
+                    {
+                        //if Both test1 and test2 run through, PlaceMark will be set
+                        PlaceMark(row, column);
+                    }
+                    else
+                    {
+                        Console.WriteLine("Invalid placement or Column! Please enter 0, 1, or 2 only.");
+                        if (playerTurn == "X")
+                        {
+                            playerTurn = "O";
+                        }
+                        else playerTurn = "X";
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Invalid placement for Row! Please enter 0, 1, or 2 only.");
+                    if (playerTurn == "X")
+                    {
+                        playerTurn = "O";
+                    }
+                    else playerTurn = "X";
+                }
+            }
+            //catch function to notify user of bad input
+            catch
+            {
+                //resets the playerTurn
+                if (playerTurn == "X")
+                {
+                    playerTurn = "O";
+                }
+                else playerTurn = "X";
+                Console.WriteLine("Invalid placement! Please enter 0, 1, or 2 only.");
+            }
         }
-
         public static void PlaceMark(int row, int column)
         {
-        // your code goes here
+            //If PlayerTurn is an X and the location is valid, set the PlaceMark
+            if (playerTurn == "X" && board[row][column] == " ")
+            {
+                board[row][column] = "X";
+
+            }
+            //If PlayerTurn is an O and the location is valid, set the PlaceMark
+            else if (playerTurn == "O" && board[row][column] == " ")
+            {
+                board[row][column] = "O";
+
+            }
+            else
+            {
+                //If PlaceMark is already taken, notify user and reset PlayerTurn
+                Console.WriteLine("Place is already taken! Place your marker somewhere else.");
+                if (playerTurn == "X")
+                {
+                    playerTurn = "O";
+                }
+                else playerTurn = "X";
```

Comment:
this is technically correct syntax, but a little hard to read. the following might be easier on the eyes
```
if(playerTurn == "X") 
{
      playerTurn = "O";
} else {
      playerTurn = "X";
}